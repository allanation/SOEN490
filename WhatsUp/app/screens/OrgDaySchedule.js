/* eslint-disable react/no-unescaped-entities */
/* eslint-disable react/jsx-key */
/* eslint-disable react/prop-types */
import React, { useState, useEffect } from "react";
import { StyleSheet, Text, View, ScrollView, Alert } from "react-native";
import Screen from "../components/Screen";
import TitleHeaders from "../components/TitleHeaders";
import colors from "../config/colors";
import AppButton from "../components/AppButton";
import ItineraryEvent from "../components/ItineraryEvent";
import UtilBtn from "../components/UtilBtn";
import AppModal from "../components/AppModal";
import AppTextInput from "../components/AppTextInput";
import { useNavigation } from "@react-navigation/native";
import uuid from "react-native-uuid";
import { storeItinerary } from "./OrgDetails";

function OrganizerDaySchedule({ route }) {
  const [schedule, setSchedule] = useState([]);
  const [modalVisible, setModalVisible] = useState(false);
  const [title, setTitle] = useState("");
  const [startTime, setStartTime] = useState("");
  const [endTime, setEndTime] = useState("");
  const [description, setDescription] = useState("");
  const [location, setLocation] = useState("");
  const navigation = useNavigation();
  const ids = uuid.v4();
  const {day, i, itinerary} = route.params;

  useEffect(() => {
    getSchedule()
}, []);

  const getSchedule = async () => {
    if(itinerary?.[i-1]){
      setSchedule(itinerary[i-1].schedule)
    }
  };

  const handleAddEvent = async (
    title,
    startTime,
    endTime,
    description,
    location
  ) => {
    if (title.length == 0) {
      Alert.alert("Error", "Please fill out the title.");
      return;
    }
    if (startTime.length == 0) {
      Alert.alert("Error", "Please fill out the start date.");
      return;
    }
    if (endTime.length == 0) {
      Alert.alert("Error", "Please fill out the end date.");
      return;
    }
    if (description.length == 0) {
      Alert.alert("Error", "Please fill out the description.");
      return;
    }
    if (location.length == 0) {
      Alert.alert("Error", "Please fill out the location.");
      return;
    }

    const newSchedule = {
      title: title,
      startTime: startTime,
      endTime: endTime,
      description: description,
      location: location,
      id: ids
    }

    setSchedule((schedule) => [...schedule, newSchedule]);
    setModalVisible(false);
  };

  const addItinerary = async (day,schedule) => {
    const newItinerary = {
      day: day.toString(),
      schedule: schedule,
    };
    if(itinerary?.[i-1]){
      itinerary[i-1] = newItinerary;
    }else{
    itinerary.push(newItinerary);
    }
  };
    

  const onRemove = (id) => () => {
    setSchedule(schedule.filter((item) => item.id !== id));
  };

  const onEdit = (newSchedule) => () => {
    setSchedule(schedule.filter((item) => item.id !== newSchedule.id));
    setSchedule((schedule) => [...schedule, newSchedule]);
  };

  const goToNextPage = async () => {
    //Store the information before leaving page
    if(i==day){
      addItinerary(i, schedule);
      storeItinerary(itinerary);
      navigation.navigate("OrgTags");
    }else{ 
      addItinerary(i, schedule);
      navigation.push("OrgDay", {day: day, i: (i+1), itinerary: itinerary});
    }
  };

  return (
    <Screen style={{ padding: 20, marginTop: 30 }}>
      <View style={{ width: "100%", display: "flex" }}>
        <View style={{ flexDirection: "row", justifyContent: "center" }}>
          <UtilBtn
            icon="chevron-back-outline"
            onPress={() => navigation.goBack()}
            style={{ position: "absolute", left: 0 }}
            testID={"backButton"}
          />
          <TitleHeaders
            style={{ alignSelf: "center" }}
            title={"Day " + (i) + " Schedule"}
          />
        </View>

        <TitleHeaders
          style={{ alignSelf: "center" }}
          isTitle={false}
          title="Please fill the following information"
        />
      </View>
      <UtilBtn
        style={{ alignSelf: "flex-end", marginRight: 24 }}
        icon="add-circle"
        onPress={() => setModalVisible(true)}
        testID={"addDayIcon"}
      />
      <ScrollView>
        <View style={{ marginTop: 12 }}>
          <View>
            {schedule.length == 0 ? (
              <Text
                style={{
                  color: colors.lightGrey,
                  marginLeft: 24,
                  marginTop: 18,
                }}
              >
                'No items in your itinerary yet...'
              </Text>
            ) : (
              schedule.map((event) => (
                <ItineraryEvent
                  title={event.title}
                  startTime={event.startTime}
                  endTime={event.endTime}
                  description={event.description}
                  location={event.location}
                  id={event.id}
                  onRemove={onRemove}
                  onEdit={onEdit}
                />
              ))
            )}
          </View>
        </View>
      </ScrollView>
      <View>
        <AppButton title={"Next"} testID={"nextButton"} onPress={() => goToNextPage()}></AppButton>
      </View>
      <AppModal
        animationType="fade"
        testID={"addDayModal"}
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <View style={styles.modalView}>
          <View style={styles.inputView}>
            <UtilBtn
              icon="chevron-back-outline"
              testID={"goBackModal"}
              style={{ position: "absolute" }}
              onPress={() => setModalVisible(!modalVisible)}
            />
            <TitleHeaders style={{ alignSelf: "center" }} title={"New Item"} />
            <ScrollView
              keyboardDismissMode="interactive"
              style={{ width: "100%" }}
            >
              <AppTextInput
                placeholder="Title"
                style={styles.shadow}
                onChangeText={(currentTitle) => setTitle(currentTitle)}
              />
              <AppTextInput
                placeholder="Start Time"
                style={styles.shadow}
                onChangeText={(currentStartTime) =>
                  setStartTime(currentStartTime)
                }
              />
              <AppTextInput
                placeholder="End Time"
                style={styles.shadow}
                onChangeText={(currentEndTime) => setEndTime(currentEndTime)}
              />
              <AppTextInput
                placeholder="Description"
                style={styles.shadow}
                onChangeText={(currentDescription) =>
                  setDescription(currentDescription)
                }
              />
              <AppTextInput
                placeholder="Location"
                style={styles.shadow}
                onChangeText={(currentLocation) => setLocation(currentLocation)}
              />
              <AppButton
                title="Add"
                style={{ marginTop: 15 }}
                testID={"addDayButton"}
                onPress={() =>
                  handleAddEvent(
                    title,
                    startTime,
                    endTime,
                    description,
                    location
                  )
                }
              />
            </ScrollView>
          </View>
        </View>
      </AppModal>
    </Screen>
  );
}

const styles = StyleSheet.create({
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 35,
    padding: 20,
    paddingTop: 20,
    width: "86%",
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  inputView: {
    marginTop: 8,
    borderColor: colors.lightGrey,
    borderRadius: 7,
    width: "90%",
    alignSelf: "center",
    paddingTop: 20,
  },
  shadow: {
    justifyContent: "center",
    shadowColor: "black", // IOS
    shadowOffset: { height: 1, width: 1 }, // IOS
    shadowOpacity: 0.2, // IOS
    shadowRadius: 3, //IOS
    elevation: 2, // Android
  },
});

export default OrganizerDaySchedule;
