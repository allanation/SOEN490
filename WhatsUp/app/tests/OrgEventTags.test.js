import "react-native";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
import OrgDaySchedule from "../screens/OrgDaySchedule";
import OrgEventTagScreen from "../screens/OrgEventTags";

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

// it("Renders Create New Event Last Page Correctly", () => {
//   const tree = render(
//     <NavigationContainer>
//       <OrgEventTagScreen />
//     </NavigationContainer>
//   ).toJSON();
//   expect(tree).toMatchSnapshot();
// });

test("Successfully adding a tag", async () => {
  render(
    <NavigationContainer>
      <OrgEventTagScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.changeText(screen.getByPlaceholderText("Ex.: University"), "Test")
  });
});

test("Successfully submitting an event", async () => {
  render(
    <NavigationContainer>
      <OrgEventTagScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("submitEvent"));
  });

});

// test("Successfully go back to org day schedule page when clicking on go back Icon", async () => {
//   render(
//     <NavigationContainer>
//       <OrgEventTagScreen />
//     </NavigationContainer>
//   );

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("backButton"));
//   });
// });

// test("Successfully create an Event when all the mandatory fields in ALL the create event pages are filled ", async () => {
//   render(
//     <NavigationContainer>
//       <OrgNewEventScreen />
//     </NavigationContainer>
//   );

//   await waitFor(() =>
//     fireEvent.changeText(screen.getByPlaceholderText("Event Title"), "Title Test")
//   );
//   await waitFor(() =>
//     fireEvent.changeText(screen.getByPlaceholderText("Organization Name"), "Organization Test")
//   );
//   await waitFor(() =>
//     fireEvent.changeText(screen.getByPlaceholderText("Location"), "Montreal")
//   );
//   await waitFor(() =>
//   fireEvent.changeText(screen.getByPlaceholderText("Description"), "Description Test")
//   );

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("chooseImage"));
//   });
  
//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("sportsImage"));
//   });

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("submitImage"));
//   });

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("nextButton"));
//   });

//   render(
//     <NavigationContainer>
//       <OrgPOCScreen />
//     </NavigationContainer>
//   );

//   await waitFor(() =>
//     fireEvent.changeText(screen.getByPlaceholderText("Name"), "Test")
//   );
//   await waitFor(() =>
//     fireEvent.changeText(screen.getByPlaceholderText("Phone Number"), "514-123-4567")
//   );
//   await waitFor(() =>
//     fireEvent.changeText(screen.getByPlaceholderText("Email"), "test@gmail.com")
//   );

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("nextButton"));
//   });

//   render(
//     <NavigationContainer>
//       <IOSDateTimePicker />
//     </NavigationContainer>
//   );

//   const startPicker = await screen.getByTestId('startTime');
//   fireEvent(startPicker, 'onChange', { nativeEvent: { timestamp: new Date('2023-03-04') } });

//   const endPicker = await screen.getByTestId('endTime');
//   fireEvent(endPicker, 'onChange', { nativeEvent: { timestamp: new Date('2023-04-04') } });

//   render(
//     <NavigationContainer>
//       <OrgDateInfoScreen />
//     </NavigationContainer>
//   );

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("nextButton"));
//   });

//   render(
//     <NavigationContainer>
//       <OrgEventTagScreen />
//     </NavigationContainer>
//   );
//   await waitFor(() => {
//     fireEvent.changeText(screen.getByPlaceholderText("Ex.: University"), "Test")
//   });

//   await waitFor(() => {
//     fireEvent.press(screen.getByTestId("submitEvent"));
//   });

// });


