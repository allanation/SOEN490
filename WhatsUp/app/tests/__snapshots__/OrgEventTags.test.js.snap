// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Create New Event Last Page Correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "marginTop": 30,
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "elevation": 2,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "marginLeft": 2,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
            "marginLeft": "1%",
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Create Event Tags
    </Text>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
            "marginLeft": "1%",
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "#747474",
            "fontSize": 20,
            "marginBottom": 6,
            "marginLeft": "1%",
            "paddingVertical": 6,
          },
        ]
      }
    >
      Please fill the following information
    </Text>
  </View>
  <RCTScrollView
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "#989797",
              "borderRadius": 12,
              "borderWidth": 1,
              "marginLeft": "1%",
              "marginVertical": 10,
              "opacity": 0.9,
              "paddingHorizontal": 12,
              "paddingVertical": 7,
              "width": "98%",
            },
            Object {
              "color": "#989797",
              "elevation": 2,
              "fontSize": 18,
              "justifyContent": "center",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize={true}
          clearButtonMode="always"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Ex.: University"
          placeholderTextColor="grey"
          style={
            Object {
              "color": "black",
              "fontFamily": "Helvetica Neue",
              "fontSize": 18,
              "opacity": 1,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Helvetica Neue",
              "marginLeft": "1%",
            },
            Object {
              "color": "gray",
              "paddingHorizontal": 20,
            },
            Object {
              "color": "#747474",
              "fontSize": 20,
              "marginBottom": 6,
              "marginLeft": "1%",
              "paddingVertical": 6,
            },
          ]
        }
      >
        Add tags to increase visibility
      </Text>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "width": "95%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#989797",
                "marginLeft": 24,
                "marginTop": 18,
              }
            }
          >
            No tags in your event yet...
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#00c0a4",
          "borderRadius": 20,
          "elevation": 4,
          "marginLeft": "1%",
          "marginVertical": 40,
          "opacity": 1,
          "padding": 12,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
          "width": "90%",
        }
      }
      testID="submitEvent"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Helvetica Neue",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
      >
        Submit Event
      </Text>
    </View>
  </View>
</View>
`;
